src/FrankenTuples.jl

```diff
@@ -418,7 +402,14 @@ julia> ftcall(mapreduce, ftuple(abs2, -, 1:4; init=0))
 -30
 ```
 """
-ftcall(f::Function, ft::FrankenTuple) = f(Tuple(ft)...; NamedTuple(ft)...)
-ftcall(f::Function, ft::FrankenTuple{Tuple{},(),Tuple{}}) = f()
+ftcall(f, ft::FrankenTuple) = f(Tuple(ft)...; NamedTuple(ft)...)
+ftcall(f, ft::FrankenTuple{Tuple{},(),Tuple{}}) = f()
+
+# NOTE: this method signature makes sure we don't define map(f)
+function Base.map(f, ft::FrankenTuple, fts::FrankenTuple...)
```

Comment:
I don't think this definition of `map` really jives with what `map` does for other collections, since iteration of `FrankenTuple`s provides the elements in sequence rather than treating the named and unnamed parts separately.

Reply:
Hm, I think that might be okay. I wonder if perhaps you expect `isequal(map(f, c), map(f, collect(c)))` (which I kind of expect as well). However that's already not the case for `NamedTuple` (second test below):

```julia
using FrankenTuples

function _test_map(f, c)
    (map(f, c), map(f, collect(c)))
end

@show _test_map(sqrt, (1, 4))
@show _test_map(sqrt, (a=9, b=16))
@show _test_map(sqrt, @ftuple (1, 4, a=9, b=16))
```

```
_test_map(sqrt, (1, 4)) = ((1.0, 2.0), [1.0, 2.0])
_test_map(sqrt, (a = 9, b = 16)) = ((a = 3.0, b = 4.0), [3.0, 4.0])
_test_map(sqrt,  @ftuple((1, 4, a = 9, b = 16))) = (FrankenTuple((1.0, 2.0), (a = 3.0, b = 4.0)), [1.0, 2.0, 3.0, 4.0])
```

I think the right "invariant" is 

```julia
function test_map_collect_commute(f, c)
    isequal(
        collect(map(f, c)),
        map(f, collect(c))
    )
end

@show test_map_collect_commute(sqrt, (1, 4))
@show test_map_collect_commute(sqrt, (a=9, b=16))
@show test_map_collect_commute(sqrt, @ftuple (1, 4, a=9, b=16))
```

and it gives

```
test_map_collect_commute(sqrt, (1, 4)) = true
test_map_collect_commute(sqrt, (a = 9, b = 16)) = true
test_map_collect_commute(sqrt, @ftuple((1, 4, a = 9, b = 16))) = true
```

Reply:
@ararslan

Reply:
Oh I see that you've just recently cherry picked some of the commits from this PR, and addressed some of the other commits too. Great!

Is `map` the only thing that this PR would add, then?

Reply:
Yep, just wanted to get the repo into a better state then discuss/ruminate on `map` separately. I made sure to preserve your authorship for things you fixed!

Reply:
I noticed, and I appreciate it :)